src/hark_lang/hark_compiler/compiler.py

```diff
@@ -301,6 +288,39 @@ def _(self, n: nodes.N_Binop):
                 ]
             )
 
+    @compile_expr.register
+    def _(self, n: nodes.N_UnaryOp):
+        if n.op == "async":
+            return self._compile_async_expr(n.rhs)
+        elif n.op == "await":
+            return self._compile_await_expr(n.rhs)
+        elif n.op == "!":
+            return self._compile_boolean_negation(n.rhs)
+        elif n.op == "-":
+            return self._compile_negation_expr(n.rhs)
+        raise ValueError(f"Unrecognised unary expression {n}")
+
+    def _compile_async_expr(self, expr: nodes.Node):
+        if not isinstance(expr, nodes.N_Call):
+            raise ValueError(f"Can't use async with {expr} - {type(expr)}")
```

Comment:
This (and the ValueError above) should probably be HarkCompileError instances, though I wrote this code originally and take full responsibility for it :P.